### PR TITLE
chore: fix ESLint warnings, use type imports

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

- Fixed some ESLint warnings detected by using config from web components repo,
- Updated most of components imports to use `import type` where it makes sense.

### CODE REVIEW

File: frontend/demo/foundation/icons-preview.ts [hunk 1]
```diff
@@ -117,8 +117,8 @@ export class IconsPreview extends HTMLElement {
 
     this.innerHTML = html;
 
-    const search = this.querySelector('input');
-    search?.addEventListener('input', () => {
+    const search = this.querySelector('input') as HTMLInputElement;
```
Comment (tomivirkki) [inline frontend/demo/foundation/icons-preview.ts:120]: nit: prefer `querySelector<HTMLInputElement>('input')` over type casting

**Review by tomivirkki — APPROVED** [review]

File: frontend/demo/foundation/icons-preview.ts (hunk 1 above)
Comment (web-padawan) [reply frontend/demo/foundation/icons-preview.ts]: Thanks, fixed.

Comment (tomivirkki) [reply frontend/demo/foundation/icons-preview.ts]: Ah, TS is smart enough to determine the type from the query 👍  Then `search?.addEventListener...` would be enough

Comment (web-padawan) [reply frontend/demo/foundation/icons-preview.ts]: Actually it was already using `search?.addEventListener` so I reverted this change as unnecessary.

**Pull request merged**